assets/JS/signupValidation.js

```diff
@@ -0,0 +1,77 @@
+import { isEmail, isNotEmpty, hasMinLength, isEqualsPassword } from './validation.js';
+
+(() => {
+  let emailInvalid = true;
+  let passwordInvalid = true;
+  let checkingPasswordInvalid = true;
+  let nickNameInvalid = true;
+  let passwordValue;
+
+  const emailInput = document.querySelector('input[type="email"]');
+  const nickNameInput = document.querySelector('input[name="nickname"]');
+  const passwordInput = document.querySelector('input[type="password"]');
+  const passwordCheckInput = document.querySelector(
+    'input[name="passwordCheck"]'
+  );
+
+  const showPasswordBtn = document.querySelectorAll(".hide-btn");
+  const passwordHideBtn = showPasswordBtn[0];
+  const checkPasswordHideBtn = showPasswordBtn[1];
+
+  function checkFormButton() {
+    const loginBtn = document.querySelector('button[type="submit"]');
+    loginBtn.disabled =
+      emailInvalid ||
+      passwordInvalid ||
+      checkingPasswordInvalid ||
+      nickNameInvalid;
+  }
+  // 회원가입 버튼 활성화
+
+  function passwordHide(btn, input) {
+    if (btn.classList[1] === undefined) {
+      input.type = "text";
+      btn.classList.add("show");
+    } else {
+      input.type = "password";
+      btn.classList.remove("show");
+    }
+  }
```

Comment:
아래처럼 toggle 을 사용해서 지금보다 더 가독성을 높이고, 조건문을 직관적으로 변경할 수도 있겠어요.

```suggestion
  function passwordHide(btn, input) {
    const isPasswordVisible = input.type === 'text';
    input.type = isPasswordVisible ? 'password' : 'text';
    btn.classList.toggle('show', !isPasswordVisible);
}
```

https://developer.mozilla.org/ko/docs/Web/API/Element/classList#toggle_string_force_